### PR TITLE
Use NSSpeechSynthesizer.AttributesForVoice to correctly populate the X.E Locale on Mac. 

### LINKED ISSUE #1622: [Bug] [Mac] TextToSpeeah Locale Object not populated correctly.
### Description

The Mac TTS Locale object only has the Language property populated and it populated incorrectly as it stores the  voice Identifier, rather than the language information. This means it is essentially useless when trying to allow your app to switch languages to speak words natively.

### Steps to Reproduce

See code here 
https://github.com/xamarin/Essentials/blob/main/Xamarin.Essentials/TextToSpeech/TextToSpeech.macos.cs#L17

### Expected Behaviour

Language property should store language info (eg fr-FR), and the other Locale properties (Name and Identifier) should be populated correctly.

### Actual Behaviour

Incorrectly populated.

### Basic Information

- Version with issue: 1.6 Stable
- Last known good version: N/A
- IDE: VS4Mac
- Platform Target Frameworks: 
  - Mac: Yes

### LINKED ISSUE #1624: [Bug] TextToSpeech sample incorrectly lists Locales and doesn't change voices on some platforms
### Description

TTS Demo doesn't behave consistently across platforms, as locales aren't populated consistently. Therefore the sample doesn't allow changing voices across all platforms.

### Steps to Reproduce

Build Samples from main. Text to Speech demo is inconsistent from iOS, Android and Mac 

### Expected Behaviour

Consistent behaviour across platforms and ability to change voices at runtime.

### Actual Behaviour

Inconsistent behaviour across platforms and inability to change voices at runtime across platforms.

### Basic Information

- Version with issue: 1.6
- Last known good version: N/A
- IDE: VS4Mac
- Platform Target Frameworks: 
  - iOS:  Latest Stable
  - Android:  Latest Stable
  - Mac:   Latest Stable

### Screenshots

Before Fix on Mac
Screenshot 2021-01-08 at 02 11 45

After Fix on Mac
Screenshot 2021-01-08 at 02 25 04

### Reproduction Link

Build from main to see the current issue.

### PR DESCRIPTION
**Pull request by CartBlanche**
### Description of Change ###

Use NSSynthesiser Attributes to correctly populate the X.E Locale properties on Mac.  
Fix Locale Name on iOS as it was incorrectly populating it with Language
Tweaked TTS demo, to list Languages, so that voices are changed correctly on all platforms.

### Bugs Fixed ###

 - Fixes #1622 - Populate Mac Locale correctly
 - Fixes #1624 - Make TTS demo behave consistently across platforms

### API Changes ###

None

Added: 
 
None

Changed:

[Mac] Pass each voice to the NSSpeechSynthesizer.AttributesForVoice(), so we can then popolate the Locale object correctly.
https://github.com/CartBlanche/Essentials/blob/dominique-FixTTSLocaleOnMac/Xamarin.Essentials/TextToSpeech/TextToSpeech.macos.cs#L15-L18

[iOS] Populate Name property correctly.
https://github.com/CartBlanche/Essentials/blob/dominique-FixTTSLocaleOnMac/Xamarin.Essentials/TextToSpeech/TextToSpeech.ios.tvos.watchos.cs#L15-L16

[Sample] Use language to populate Action sheet, so languages can be changes across platforms.
https://github.com/CartBlanche/Essentials/blob/dominique-FixTTSLocaleOnMac/Samples/Samples/ViewModel/TextToSpeechViewModel.cs#L91-L95

### Behavioural Changes ###

Tweaked how the Locale on Mac is populated as per changes above.

### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description)
- [x] Fixed samples (if omitted, state reason in description)
- [x] Rebased on top of `main` at time of PR
- [x] Changes adhere to coding standard
- [ ] Updated documentation ([see walkthrough](https://github.com/xamarin/Essentials/wiki/Documenting-your-code-with-mdoc))

### CODE REVIEW

**Comment by CartBlanche** [comment]
@mattleibow, @jamesmontemagno Let me know if there is anything else I should do to this PR, to get this merged??

File: Xamarin.Essentials/TextToSpeech/TextToSpeech.macos.cs [hunk 1]
```diff
@@ -14,7 +14,8 @@ public static partial class TextToSpeech
 
         internal static Task<IEnumerable<Locale>> PlatformGetLocalesAsync() =>
             Task.FromResult(NSSpeechSynthesizer.AvailableVoices
-                .Select(v => new Locale(v, null, null, null)));
+                .Select(voice => NSSpeechSynthesizer.AttributesForVoice(voice))
+                .Select(attribute => new Locale(attribute["VoiceLanguage"].ToString(), null, attribute["VoiceName"].ToString(), attribute["VoiceIdentifier"].ToString())));
```
Comment (mattleibow) [inline Xamarin.Essentials/TextToSpeech/TextToSpeech.macos.cs:18]: Is there a chance for any of these to be null? Or the keys not exist?

File: Xamarin.Essentials/TextToSpeech/TextToSpeech.macos.cs [hunk 2]
```diff
@@ -30,7 +31,7 @@ internal static async Task PlatformSpeakAsync(string text, SpeechOptions options
                         ss.Volume = NormalizeVolume(options.Volume);
 
                     if (options.Locale != null)
-                        ss.Voice = options.Locale.Language;
+                        ss.Voice = options.Locale.Id;
```
Comment (mattleibow) [inline Xamarin.Essentials/TextToSpeech/TextToSpeech.macos.cs:34]: 🤦

**Review by mattleibow — COMMENTED** [review]
Looks good. Just checking on those possible nulls. If they will never be, then fine. But if possible, maybe there should be some check.

File: Xamarin.Essentials/TextToSpeech/TextToSpeech.macos.cs (hunk 1 above)
Comment (CartBlanche) [reply Xamarin.Essentials/TextToSpeech/TextToSpeech.macos.cs:18]: @mattleibow The docs (https://developer.apple.com/documentation/appkit/nsspeechsynthesizer/1448458-attributesforvoice?language=objc) don't mention anything, but you are right a `?` before the `ToString()` call wouldn't hurt and could stop some breakages.  Unless you have another suggestion for it?

Comment (CartBlanche) [reply Xamarin.Essentials/TextToSpeech/TextToSpeech.macos.cs]: @mattleibow ? Checks added to latest commit.

**Review by CartBlanche — COMMENTED** [review]
Nice!

**Comment by mattleibow** [comment]
I did some testing on Android and they had a bunch of en for all the variants. I think we need to at some point parse the languages and create a better locale object.

**Comment by CartBlanche** [comment]
@mattleibow When will this his X.E preview nugets?

**Pull request merged**